client/js/widgets/holder.js

```diff
@@ -35,10 +37,22 @@ class Holder extends Widget {
     return children.filter(w=>{
       if(acceptPiles && w.get('type') == 'pile')
         return true;
-      for(const p in this.get('dropTarget'))
-        if(w.get(p) != this.get('dropTarget')[p])
-          return false;
-      return true;
+      const dropTarget = asArray(this.get('dropTarget'))
+      let isValid = false;
+      for(let i = 0; i < dropTarget.length; i++) {
+        let isValidObject = true;
+        for(const p in dropTarget[i]) {
+          if(w.get(p) !=(dropTarget[i])[p]) {
+            isValidObject = false;
+            break;
+          }
+        }
+        if(isValidObject) {
+          isValid = true;
+          break;
+        }
+      }
+      return isValid;
```

Comment:
Feels like this code is very similar to the other code. It should probably be put into a function in `main.js`?